api/companies/destroy.js

```diff
@@ -0,0 +1,15 @@
+let router = require('../../middlewares/router')
+const db = require('../../db')
+
+router.del('/companies/:id', function *(next) {
+  if(!!db.companies[this.params.id]) {
+    if(delete(db.companies[this.params.id])) {
+      this.response.status = 204
+    } else {
+      this.response.status = 500
```

Comment:
500 errors should not be set manually. Why you respond with 500 at this pretty normal case. Can't delete? - fine – it's 200.